code/game/objects/structures/holosign.dm

```diff
@@ -180,6 +180,39 @@
 	else
 		return ..()
 
+/obj/structure/holosign/barrier/detective
+	name = "investigation barrier"
+	desc = "An authoritive holographic barrier proclaiming a crime scene. Energy arcs off of it."
+	max_integrity = 20
+	icon_state = "holosign_det"
+	var/shockcd = 0
+
+/obj/structure/holosign/barrier/detective/bullet_act(obj/projectile/P)
+	take_damage(P.damage, BRUTE, MELEE, 1)
+	return BULLET_ACT_HIT
+
+/obj/structure/holosign/barrier/detective/attack_hand(mob/living/user)
+	. = ..()
+	if(.)
+		return
+	if(shockcd || !isliving(user))
+		return
+	user.electrocute_act(10,"Energy Barrier", flags = SHOCK_NOGLOVES)
+	shockcd = TRUE
+	addtimer(CALLBACK(src, PROC_REF(cooldown)), 5 SECONDS)
+
+/obj/structure/holosign/barrier/detective/Bumped(atom/movable/victim)
+	if(shockcd || !isliving(victim))
+		return
+	var/mob/living/living_victim = victim
+	living_victim.Knockdown(10)
+	living_victim.electrocute_act(15,"Energy Barrier", flags = SHOCK_NOGLOVES | SHOCK_NOSTUN)
+	shockcd = TRUE
+	addtimer(CALLBACK(src, PROC_REF(cooldown)), 5 SECONDS)
+
+/obj/structure/holosign/barrier/detective/proc/cooldown()
+	shockcd = FALSE
```

Comment:
Use a cooldown timer instead of add timer. You don't need to have a timer with a proc call if you are just checking to see if the time has elapsed since since last activation. See the cooldown defines.